src/Microsoft.IdentityModel.Tokens/Encryption/EcdhKeyExchangeProvider.cs

```diff
@@ -56,51 +52,39 @@ public class EcdhKeyExchangeProvider //todo: rename to KeyExchangeProvider, or E
         /// <summary>
         /// Initializes a new instance of <see cref="EcdhKeyExchangeProvider"/> used for CEKs
         /// <param name="privateKey">The <see cref="ECDsaSecurityKey"/> that will be used for cryptographic operations and represents the private key.</param>
-        /// <param name="publicKey">The <see cref="ECDsaSecurityKey"/> that will be used for cryptographic operations and represents the public key.</param>
+        /// <param name="publicKey">The <see cref="SecurityKey"/> that will be used for cryptographic operations and represents the public key.</param>
         /// <param name="alg">alg header parameter value.</param>
         /// <param name="enc">enc header parameter value.</param>
         /// </summary>
-        public EcdhKeyExchangeProvider(ECDsaSecurityKey privateKey, ECDsaSecurityKey publicKey, string alg, string enc)
+        public EcdhKeyExchangeProvider(ECDsaSecurityKey privateKey, SecurityKey publicKey, string alg, string enc)
         {
             if (privateKey == null)
                 throw LogHelper.LogArgumentNullException(nameof(privateKey));
-            if (publicKey is null)
-                throw LogHelper.LogArgumentNullException(nameof(publicKey));
-
-            ValidateAlgAndEnc(alg, enc);
-
-            SetKeyDataLenAndEncryptionAlgorithm(alg, enc);
-            _ecParamsPrivate = privateKey.ECDsa.ExportParameters(true);
-            _ecParamsPublic = publicKey.ECDsa.ExportParameters(false);
-
-            ValidateCurves(nameof(privateKey), nameof(publicKey));
 
-            Initialize();
-        }
-
-        /// <summary>
-        /// Initializes a new instance of <see cref="EcdhKeyExchangeProvider"/> used for CEKs
-        /// <param name="privateKey">The <see cref="ECDsaSecurityKey"/> that will be used for cryptographic operations and represents the private key.</param>
-        /// <param name="publicKey">The <see cref="JsonWebKey"/> that will be used for cryptographic operations and represents the public key.</param>
-        /// <param name="alg">alg header parameter value.</param>
-        /// <param name="enc">enc header parameter value.</param>
-        /// </summary>
-        public EcdhKeyExchangeProvider(ECDsaSecurityKey privateKey, JsonWebKey publicKey, string alg, string enc)
-        {
-            if (privateKey == null)
-                throw LogHelper.LogArgumentNullException(nameof(privateKey));
             if (publicKey is null)
                 throw LogHelper.LogArgumentNullException(nameof(publicKey));
 
-            ValidateAlgAndEnc(alg, enc);
+            if (publicKey is ECDsaSecurityKey ecdsaKey)
```

Comment:
You should call [TryConvertToECDsaSecurityKey](https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/blob/050297acc5b1dacba8f706459af6da8803762453/src/Microsoft.IdentityModel.Tokens/JsonWebKeyConverter.cs#L344)

You could write this as:
```cs
if (!TryConvertToECDsaSecurityKey(publicKey, out ecdaPublickey)
    throw LogHelper.LogArgumentException<ArgumentException>(
                    nameof(publicKey),
                    LogHelper.FormatInvariant(LogMessages.IDX11003, LogHelper.MarkAsNonPII(publicKey.GetType())));

    // then see if this works... if so, then GetEcParamsFromJwk would not be needed.
    _ecParamsPublic = ecdsaPublickey.ECDSA.ExportParameters(false);
```